wrong usage of assert

Some of the tests will never fail, because `assert` is used wrong.
The API of `assert` is: `assert(value, message)` but in the tests it was used like this `assert(actual, expected)` (checkout http://nodejs.org/api/assert.html).

When I changed some of the tests to `assert.deepEqual` they failed.
I will submit a PR to fix the tests.

The failing test was:
```
1) selectionsort node module should work with sorted arrays:
     AssertionError: [-5,-5,-5,-5,2,2,12,14] deepEqual [-5,1,2,5,12,12,14,16]
```

--- a/index.js
+++ b/index.js
@@ -27,10 +27,11 @@ module.exports = function (array, compareFunction) {
         min = array[j];
         index = j;
       }
-      temp = array[i];
-      array[i] = min;
-      array[index] = temp;
     }
+
+    temp = array[i];
+    array[i] = min;
+    array[index] = temp;
   }
 
   return array;

--- a/test/test.js
+++ b/test/test.js
@@ -19,13 +19,13 @@ function makeRandomArray() {
 describe('selectionsort node module', function () {
 
   it('should work with empty arrays', function () {
-    assert(selectionsort([]), []);
+    assert.deepEqual(selectionsort([]), []);
   });
 
   it('should work with sorted arrays', function () {
-    assert(selectionsort([1, 2, 3, 4]), [1, 2, 3, 4]);
-    assert(selectionsort([5, 1, 12, -5, 16, 2, 12, 14]), [-5, 1, 2, 5, 12, 12, 14, 16]);
-    assert(selectionsort([6, 4, 9, 3, 1, 7]), [1, 3, 4, 6, 7, 9]);
+    assert.deepEqual(selectionsort([1, 2, 3, 4]), [1, 2, 3, 4]);
+    assert.deepEqual(selectionsort([5, 1, 12, -5, 16, 2, 12, 14]), [-5, 1, 2, 5, 12, 12, 14, 16]);
+    assert.deepEqual(selectionsort([6, 4, 9, 3, 1, 7]), [1, 3, 4, 6, 7, 9]);
   });
 
   it('should work with random non-sorted arrays', function () {

